src/libcrun/container.c

```diff
@@ -2504,9 +2504,21 @@ libcrun_container_exec (libcrun_context_t *context, const char *id, runtime_spec
         }
 
       if (context->kontain) {
-        int rc = libcrun_kontain_argv(&process->args, &exec_path);
+        char* new_execpath;
+        int rc = libcrun_kontain_nonkmexec_allowed(exec_path, &new_execpath);
         if (rc != 0) {
-          libcrun_fail_with_error (rc, "exec: fixup argv");
+          libcrun_fail_with_error(rc, "allow non-km executable check failed");
+        }
+        if (new_execpath == NULL) {
+          // Run as a payload
+          rc = libcrun_kontain_argv(&process->args, &exec_path);
+          if (rc != 0) {
+            libcrun_fail_with_error (rc, "exec: fixup argv");
```

Comment:
what does this message mean ? I would be super confused if I saw this

Reply:
It means there was an error while tryng to run the container entry point as a km payload.
Is this better:

libcrun_fail_with_error (rc, "failed to setup %s to run in a kontain VM", exec_path);